pkgs/development/ocaml-modules/directories/default.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, stdenv, fetchFromGitHub, buildDunePackage, ctypes }:
+
+buildDunePackage rec {
+  pname = "directories";
+  version = "0.2";
+  useDune2 = true;
+
+  minimumOCamlVersion = "4.07";
+
+  src = fetchFromGitHub {
+    owner = "ocamlpro";
+    repo = pname;
+    rev = version;
+    sha256 = "0s7ginh0g0fhw8xf9v58cx99a8q9jqsf4i0p134m5qzf84qpjwff";
+  };
+
+  buildInputs = lib.optional (stdenv.hostPlatform.isWindows or stdenv.hostPlatform.isCygwin) ctypes;
```

Comment:
I think this should be `targetPlatform` rather than `hostPlatform`. Not too relevant atm, but in case we can cross compile OCaml in the future this would be necessary.


The opam file also lists `ctypes-foreign`? Is that not necessary?

Reply:
> 
> 
> I think this should be `targetPlatform` rather than `hostPlatform`. Not too relevant atm, but in case we can cross compile OCaml in the future this would be necessary.

Thanks, fixed.
> 
> The opam file also lists `ctypes-foreign`? Is that not necessary?

At a glance, it seems `ctypes-foreign` is provided by `ocamlPackages.ctypes`, but I don't have a Windows machine with nix to test this on:

https://github.com/ocamllabs/ocaml-ctypes/blob/master/ctypes-foreign.opam
> 
> `ctypes-foreign` is just a virtual OPAM package that determines
> whether the foreign subpackage should built as part of ctypes.
> In order to actually get the ctypes package, you should also:
> 
>     opam install ctypes ctypes-foreign
> 
> You can verify the existence of the ocamlfind subpackage by:
> 
>     ocamlfind list | grep ctypes
> 

```
$ nix-shell -p ocamlPackages.ctypes ocamlPackages.findlib --run 'ocamlfind list | grep ctypes'
ctypes              (version: 0.17.1)
ctypes.foreign      (version: 0.17.1)
ctypes.foreign.base (version: 0.17.1)
ctypes.foreign.threaded (version: 0.17.1)
ctypes.foreign.unthreaded (version: 0.17.1)
ctypes.stubs        (version: 0.17.1)
ctypes.top          (version: 0.17.1)
```

Reply:
You could also just leave out the window-specific stuff for now if you can't test it since `ocaml` doesn't evaluate on window, I think:

```
nix-repl> (import <nixpkgs> {}).ocaml.meta.platforms
[ "aarch64-linux" "armv5tel-linux" "armv6l-linux" "armv7a-linux" "armv7l-linux" "mipsel-linux" "i686-linux" "x86_64-linux" "powerpc64-linux" "powerpc64le-linux" "riscv32-linux" "riscv64-linux" "x86_64-darwin" "i686-darwin" "aarch64-darwin" "armv7a-darwin" ]
```

Reply:
Makes sense.  Rebased and squashed.